jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildTarTask.java

```diff
@@ -0,0 +1,182 @@
+import org.gradle.api.tasks.options.Option;
+
+public class BuildTarTask extends DefaultTask {
+
+  /** {@code User-Agent} header suffix to send to the registry. */
+  private static final String USER_AGENT_SUFFIX = "jib-gradle-plugin";
+
+  private static final HelpfulSuggestions HELPFUL_SUGGESTIONS =
+      HelpfulSuggestionsProvider.get("Building image tarball failed");
+
+  @Nullable private JibExtension jibExtension;
+
+  /**
+   * This will call the property {@code "jib"} so that it is the same name as the extension. This
+   * way, the user would see error messages for missing configuration with the prefix {@code jib.}.
+   *
+   * @return the {@link JibExtension}.
+   */
+  @Nested
+  @Nullable
+  public JibExtension getJib() {
+    return jibExtension;
+  }
+
+  /**
+   * The target image can be overridden with the {@code --image} command line option.
+   *
+   * @param targetImage the name of the 'to' image.
+   */
+  @Option(option = "image", description = "The image reference for the target image")
+  public void setTargetImage(String targetImage) {
+    Preconditions.checkNotNull(jibExtension).getTo().setImage(targetImage);
+  }
+
+  /**
+   * @return the input files to this task are all the output files for all the dependencies of the
+   *     {@code classes} task.
+   */
+  @InputFiles
+  public FileCollection getInputFiles() {
+    return GradleProjectProperties.getInputFiles(getProject());
+  }
+
+  /**
+   * The output file to check for task up-to-date.
+   *
+   * @return the output path
+   */
+  @OutputFile
+  public String getOutputDirectory() {
+    return getTargetPath();
+  }
+
+  /**
+   * Returns the output directory for the tarball. By default, it is {@code build/jib.tar}.
+   *
+   * @return the output directory
+   */
+  private String getTargetPath() {
+    return getProject().getBuildDir().toPath().resolve("jib.tar").toString();
+  }
+
+  @TaskAction
+  public void buildTar() throws InvalidImageReferenceException, IOException {
+    // Asserts required @Input parameters are not null.
+    Preconditions.checkNotNull(jibExtension);
+    GradleBuildLogger gradleBuildLogger = new GradleBuildLogger(getLogger());
+    jibExtension.handleDeprecatedParameters(gradleBuildLogger);
+
+    RegistryCredentials knownBaseRegistryCredentials = null;
+    Authorization fromAuthorization = jibExtension.getFrom().getImageAuthorization();
+    if (fromAuthorization != null) {
+      knownBaseRegistryCredentials = new RegistryCredentials("jib.from.auth", fromAuthorization);
+    }
+
+    GradleProjectProperties gradleProjectProperties =
+        GradleProjectProperties.getForProject(getProject(), gradleBuildLogger);
+    String mainClass = gradleProjectProperties.getMainClass(jibExtension);
+    ImageReference targetImage =
+        gradleProjectProperties.getDockerTag(jibExtension, gradleBuildLogger);
+
+    // Builds the BuildConfiguration.
+    // TODO: Consolidate with BuildImageTask/BuildDockerTask.
```

Comment:
This method does seem to have a lot of what I expected would be common to several of our tasks.